dash/dash-renderer/src/utils/libraries.ts

```diff
@@ -0,0 +1,34 @@
+type LibraryResource = {
+    type: '_js_dist' | '_css_dist';
+    url: string;
+};
+
+export function loadLibrary(resource: LibraryResource) {
+    let prom;
+    const head = document.querySelector('head');
+    if (resource.type === '_js_dist') {
+        const element = document.createElement('script');
+        element.src = resource.url;
+        element.async = true;
+        prom = new Promise<void>((resolve, reject) => {
+            element.onload = () => {
+                resolve();
+            };
+            element.onerror = error => reject(error);
+        });
```

Comment:
since the way you are creating the promise is the same for both, you could extract that into a function to avoid duplication. Like with a `makePromise(element)` which would do that.